libnavigation-core/src/test/java/com/mapbox/navigation/core/routerefresh/RouteRefreshControllerTest.kt

```diff
@@ -47,10 +47,10 @@ import java.util.concurrent.TimeUnit
 class RouteRefreshControllerTest {
 
     private val logger = mockk<LoggerFrontend>(relaxed = true)
-    private val currentIndicesSnapshot = CurrentIndicesSnapshot(0, 1, 2)
+    private val currentIndicesSnapshot = CurrentIndicesSnapshotFactory.createIndices(0, 1, 2)
     private val currentIndicesSnapshotProvider =
         mockk<CurrentIndicesSnapshotProvider>(relaxed = true) {
-            coEvery { getFilledIndicesAndFreeze() } returns currentIndicesSnapshot
+            coEvery { getFilledIndicesOrWait() } returns currentIndicesSnapshot
```

Comment:
Is there a test which verifies if route refresh controller waits for route progress longer than interval between refresh tries?

Reply:
I'm not sure I follow. How is that connected? We first wait the refresh timeout, then wait for the indices.

Reply:
> We first wait the refresh timeout, then wait for the indices.

yes, but what happens if the navigator won't receive indices in 4x time of route refresh? Will route refresh controller invalidate traffic annotations?

Reply:
It either receives indices or freezes, isn't it? It can't "not receive indices".